### PR TITLE
feat: [M3-6525] - Storybook config cleanup

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
This PR is the first step in getting Storybook in a better shape. This PR addresses base config and several stability improvements. It paves the way for enabling a POC to demonstrate the way we want to build stories with storybook 7 and autodocs. Please refer to the self review for more details about this change set.

## How to test 🧪
**Storybook**
1. Run `yarn storybook` and navigate through our stories - notice console warnings gone and everything behaving as it should - make sure assets are properly served (fonts)
2. Run `yarn build-storybook` and serve the build - make sure no build error or issue, make sure assets are properly served (fonts)

### CODE REVIEW

**Review by bnussman-akamai — COMMENTED** [review]
Thanks for switching out `storybook-dark-mode-v7`! Things generally look good, but after having storybook open for a few minutes, my dev server goes crazy. I can't tell if this is a problem caused by your changed or just a general Storybook + Vite problem 

Screenshot 2023-04-26 at 2 08 18 PM

File: packages/manager/.storybook/main.ts [hunk 1]
```diff
@@ -10,16 +10,21 @@ const config: StorybookConfig = {
...
   staticDirs: ['../public'],
   framework: {
     name: '@storybook/react-vite',
     options: {},
   },
   features: { storyStoreV7: true },
+  docs: {
+    autodocs: true,
+    defaultName: 'Documentation',
+  },
```
[the remark this answers is not in the document]
Comment (jdamore-linode) [reply packages/manager/.storybook/main.ts:24]: Does this mean that our docs in Storybook will be generated using JSDoc/TSDoc doc comments in the codebase?

[the remark this answers is not in the document]
Comment (abailly-akamai) [reply packages/manager/.storybook/main.ts:24]: Yes - granted we add them :) - I'll demonstrate that in the POC, but I believe we want to get away from writing prop descriptions in storybook and move them over to our interfaces as JSdoc. And yes it gets auto-generated that way which is pretty sweet

**Comment by abailly-akamai** [comment]
@bnussman-akamai ugh! this is an existing issue I was trying to fix with this PR. I was hopeful cause I did not encounter it again with this PR. Did you `rm -rf node_modules` when testing this by any chance?

**Comment by bnussman-akamai** [comment]
Cleared my vite cache (`rm -rf node_modules/.vite`) and things were much more stable! Thanks. Consider my comment a false alarm 🚨

File: packages/manager/.storybook/manager-head.html [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<style>
+   #storybook-preview-iframe {
+    background-color: transparent;
+   }
+</style>
```
Comment (bnussman-akamai) [inline packages/manager/.storybook/manager-head.html:5]: Can help me see what change this makes? I can't seem to find the difference it makes.

Comment (abailly-akamai) [reply packages/manager/.storybook/manager-head.html:5]: The preview iframe has a background color set that prevents dark mode from showing (currently in develop).

ex: /?path=/story/elements-button--primary (dark mode)

Screenshot 2023-04-27 at 10 25 41 AM

Comment (jaalah-akamai) [reply packages/manager/.storybook/manager-head.html:5]: I changed color to red to highlight area
Screen Shot 2023-04-27 at 10 29 02 AM

Comment (abailly-akamai) [reply packages/manager/.storybook/manager-head.html:5]: I was really hoping there was a better way to configure the iframe styles but couldn't find any

**Review by bnussman-akamai — APPROVED** [review]
🚀

**Pull request merged**